vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -319,6 +317,59 @@ public void close() {
     return this;
   }
 
+  @Override
+  public io.vertx.ext.mongo.MongoClient distinct(String collection, String fieldName, String resultClassname, Handler<AsyncResult<JsonArray>> resultHandler) {
+    requireNonNull(collection, "collection cannot be null");
+    requireNonNull(fieldName, "fieldName cannot be null");
+    requireNonNull(resultHandler, "resultHandler cannot be null");
+
+    MongoCollection<JsonObject> coll = getCollection(collection);
+    try {
+      DistinctIterable distinctValues = coll.distinct(fieldName, Class.forName(resultClassname));
+      List results = new ArrayList();
+      distinctValues.into(results, (result, error) -> {
+        vertx.runOnContext(v -> {
+          if (error != null) {
+            resultHandler.handle(Future.failedFuture(error));
+          } else {
+            resultHandler.handle(Future.succeededFuture(new JsonArray((List) result)));
+          }
+        });
+      });
+    } catch (Exception e) {
+      resultHandler.handle(Future.failedFuture(e));
+    }
+    return this;
+  }
+
+  @Override
+  public MongoClient distinctBatch(String collection, String fieldName, String resultClassname, Handler<AsyncResult<JsonObject>> resultHandler) {
+    requireNonNull(collection, "collection cannot be null");
```

Comment:
There's a bit of duplicate code in this method with the distinct method - perhaps a common refactoring could be applied here.